tty/usbacm/usbacm.c

```diff
@@ -302,62 +367,114 @@ static usbacm_dev_t *usbacm_devAlloc(void)
 }
 
 
-static int _usbacm_open(usbacm_dev_t *dev, int flags)
+static int _usbacm_urbsAlloc(usbacm_dev_t *dev)
 {
 	int i;
-	fifo_t *fifo;
 
-	if (dev->flags != 0)
-		return -EIO;
+	dev->urbctrl = usb_urbAlloc(dev->pipeCtrl, NULL, usb_dir_out, 0, usb_transfer_control);
+	if (dev->urbctrl < 0) {
+		return -1;
+	}
 
-	if (dev->fifo == NULL) {
-		if ((fifo = malloc(sizeof(fifo_t) + RX_FIFO_SIZE)) == NULL)
-			return -ENOMEM;
+	for (i = 0; i < USBACM_N_URBS; i++) {
+		dev->urbs[i] = usb_urbAlloc(dev->pipeBulkIN, NULL, usb_dir_in, USBACM_BULK_SZ, usb_transfer_bulk);
+		if (dev->urbs[i] < 0) {
+			while (--i >= 0) {
```

Comment:
Two MISRA rules violated:
- modification of interator inside for loop
- condition with side-effect